src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -304,26 +299,52 @@ private async Task ProcessServiceChange(WatchEventType watchEventType, Service s
                             var projectViewModel = ConvertToProjectViewModel(
                                 _applicationModel, _servicesMap.Values, _endpointsMap.Values, executable);
 
-                            await _projectViewModelChangesChannel.Writer.WriteAsync(
-                                new ResourceChanged<ProjectViewModel>(ObjectChangeType.Modified, projectViewModel), _cancellationToken)
-                                .ConfigureAwait(false);
+                            await WriteProjectChange(projectViewModel).ConfigureAwait(false);
                         }
                         else
                         {
                             // Executable
                             var executableViewModel = ConvertToExecutableViewModel(
                                 _applicationModel, _servicesMap.Values, _endpointsMap.Values, executable);
 
-                            await _executableViewModelChangesChannel.Writer.WriteAsync(
-                                new ResourceChanged<ExecutableViewModel>(ObjectChangeType.Modified, executableViewModel), _cancellationToken)
-                                .ConfigureAwait(false);
+                            await WriteExecutableChange(executableViewModel).ConfigureAwait(false);
                         }
                     }
                     break;
             }
         }
     }
 
+    private async Task WriteProjectChange(ProjectViewModel projectViewModel, ObjectChangeType changeType = ObjectChangeType.Modified)
+    {
+        await _projectViewModelChangesChannel.Writer.WriteAsync(
+            new ResourceChanged<ProjectViewModel>(changeType, projectViewModel), _cancellationToken)
```

Comment:
can create resourechanged only once and reuse.

Reply:
The generic type is different (`ResourceChanged<ProjectViewModel>` vs `ResourceChanged<ResourceViewModel>`) so we can't do that. I guess maybe we could add an IResourceChanged and use generic variance to make it work? But that seemed outside the scope.